src/hotspot/share/classfile/resolutionErrors.hpp

```diff
@@ -56,42 +49,40 @@ class ResolutionErrorTable : AllStatic {
   // RedefineClasses support - remove obsolete constant pool entry
   static void delete_entry(ConstantPool* c);
 
+  // This value is added to the cpCache index of an invokedynamic instruction when
+  // storing the resolution error resulting from that invokedynamic instruction.
+  // This prevents issues where the cpCache index is the same as the constant pool
+  // index of another entry in the table.
+  static const int CPCACHE_INDEX_MANGLE_VALUE = 1000000;
+
```

Comment:
Do we still need `CPCACHE_INDEX_MANGLE_VALUE` and `encode_cpcache_index(int index)`? The comment around that function says that `index` is negative, so it shouldn't be confused with regular cp indices which are non-negative.

Reply:
I wanted to leave that in so that there's an assert that it's negative first.  The whole cpCache index thing is something that @matias9927 is trying to rationalize with his ResolvedIndyInfo array - should we continue to use negative indexes in the bytecode stream or not?  We think not so he'll clean this part up, and if we do we'll need some encoding like what's in CPCACHE_INDEX_MANGLE_VALUE.  Maybe the mangle value should be negative and he can change his positive indexes to negative to distinguish them from constant pool indexes.

Reply:
> I wanted to leave that in so that there's an assert that it's negative first. The whole cpCache index thing is something that @matias9927 is trying to rationalize with his ResolvedIndyInfo array - should we continue to use negative indexes in the bytecode stream or not? We think not so he'll clean this part up, and if we do we'll need some encoding like what's in CPCACHE_INDEX_MANGLE_VALUE. Maybe the mangle value should be negative and he can change his positive indexes to negative to distinguish them from constant pool indexes.

OK, we can revisit this later.